### DISCUSSION TITLE
Go 1.22 Enhanced Routing Patterns Question

### DISCUSSION THREAD

**Discussion by conways-glider** [oapi-codegen/oapi-codegen#1464] [Ideas]
With the release of Go 1.22, the http.ServeMux type [now accepts patterns and methods for the routes passed to a http.ServeMux](https://tip.golang.org/doc/go1.22#enhanced_routing_patterns). Should we consider adding a proper and first class `net/http` generation?

**Comment by jamietanna** [comment]
Yes! It's already planned (https://github.com/deepmap/oapi-codegen/issues/1068) and I started working on it when I had a bit of free time, but to be honest I need to focus on doing some triage + supporting the outstanding issues we've got then I can try and get back to this.

I'm looking forward to playing with how it works and how we can add it into oapi-codegen, then I'll also look at making sure the "adding a new server" is documented better, too

**Reply by conways-glider** [reply]
Thank you for the response! I appreciate it!

**Reply by jamietanna** [reply]
Thanks to @thedadams this support is now live but **not yet released**. Feel free to play around wit hit by pinning to the latest commit(s) rather than the latest tag - hopefully another release will be out this month :crossed_fingers:

**Reply by jamietanna** [reply]
This is now released :rocket:

**Reply by conways-glider** [reply]
Thank you so much!
